### PR TITLE
Swaps: Input List

### PR DESCRIPTION
**Pull request by benisgold**
Fixes APP-####

### CODE REVIEW

File: src/__swaps__/screens/Swap/components/CoinRow2.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+import Animated from 'react-native-reanimated';
+import { StyleSheet } from 'react-native';
+import { useSwapContext } from '@/__swaps__/screens/Swap/providers/swap-provider';
+import { UniqueId } from '@/__swaps__/types/assets';
+import { userAssetsStore } from '@/state/assets/userAssets';
+
+export const CoinRow2 = ({ assetId, output, onPress }: { assetId: string; output?: boolean; onPress: (assetId: UniqueId) => void }) => {
+  const { AnimatedSwapStyles } = useSwapContext();
+
+  const asset = userAssetsStore(state => state.getUserAsset(assetId));
```
Comment (walmat) [inline src/__swaps__/screens/Swap/components/CoinRow2.tsx:16]: instead of subscribing to the whole asset here, we should create `Cell` components for each piece of data and only subscribe to that data inside of that Cell.

File: src/__swaps__/screens/Swap/components/CoinRow2.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  const isFavorited = userAssetsStore(state => state.isFavorite(asset.address));
+  const toggleFavorite = userAssetsStore(state => state.toggleFavorite);
```
Comment (walmat) [inline src/__swaps__/screens/Swap/components/CoinRow2.tsx:18]: Instead of subscribing to this as well, we should inline it, see next comment.

File: src/__swaps__/screens/Swap/components/CoinRow2.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+                )}
+              </Inline>
+            </Stack>
+          </Inline>
+          {output ? (
+            <Inline space="8px">
+              <CoinRowButton icon="􀅳" outline size="icon 14px" />
+              <CoinRowButton
+                color={isFavorited ? '#FFCB0F' : undefined}
+                onPress={() => toggleFavorite(asset.address)}
```
Comment (walmat) [inline src/__swaps__/screens/Swap/components/CoinRow2.tsx:90]: As mentioned before, we should inline like `userAssetsStore.getState().toggleFavorite(asset.uniqueId)`

File: src/__swaps__/screens/Swap/components/UserAssetsSync.tsx [hunk 4]
```diff
@@ -29,8 +29,10 @@ export const UserAssetsSync = () => {
         data.forEach(asset => {
           if (filter === 'all' || asset.chainId === filter) {
             if (searchQuery) {
-              const stringToSearch = `${asset.name} ${asset.symbol} ${asset.address}`.toLowerCase();
-              if (stringToSearch.includes(searchQuery)) {
+              const nameMatch = asset.name.toLowerCase().includes(searchQuery);
+              const symbolMatch = asset.symbol.toLowerCase().startsWith(searchQuery);
+              const addressMatch = asset.address.toLowerCase().startsWith(searchQuery);
+              if (nameMatch || symbolMatch || addressMatch) {
```
Comment (walmat) [inline src/__swaps__/screens/Swap/components/UserAssetsSync.tsx:35]: why the change here?

Comment (walmat) [reply src/__swaps__/screens/Swap/components/UserAssetsSync.tsx:35]: oh hmm i see nvm, good catch

Comment (benisgold) [reply src/__swaps__/screens/Swap/components/UserAssetsSync.tsx:35]: we can prob improve later also. for example, rank `startsWith` matches for name higher than `includes` matches for name, if that makes sense.

**Review by walmat — APPROVED** [review]
